### PR TITLE
docs: Rework structure and few improvements

### LINKED ISSUE #1184: [EPIC] docs: Improve structure
This issue tracks a documentation restructuring for Inspektor Gadget. Image-based gadgets are now the first focus of Inspektor Gadget and the documentation doesn't reflect it very clearly. There is also some ongoing work on different PRs and issues but we need to define the structure to keep it aligned. 

# Related Issues

- #2151
- #2374
- #2727

# Proposal 

This is a proposal for the documentation structure of Inspektor Gadget

### Readme
- Badges
- Quick Intro
- Link to website

### Contributing
- setup dev environment
- tests
- ...

## Website

Published on https://www.inspektor-gadget.io/docs
Docs stored in https://github.com/inspektor-gadget/inspektor-gadget/tree/main/docs

### Intro 

Landing page of https://www.inspektor-gadget.io/docs

- What is Inspektor Gadget? (Intro) 
- gif demo (single or multiple gadgets TBD)
- Briefly describe the core concepts (then add link to full documentation)
  - Gadgets
  - Enrichment 
  - Filtering 
  - APIs 
- Environments where it can work (introduction with few details)
  - Kubernetes (kubectl-gadget) 
  - linux hosts (ig)
 - Links to:
   - requirements
   - architecture
   - contributing

### Getting started 
- Installing on Kubernetes
- Installing on Linux
- Installing on Mac? 
- Using a Gadget

### Core concepts

(#3229)

- Gadgets 
  - Link "official-gadgets"
- Architecture 
- Operators
  - Concept
  - Initialization ("optional" behavior)
  - Existing operators (bpf, wasm, btfgen, localmanager, kubemanager, etc)
- Enrichment 
- User space processing 
- Data Export
  - Prometheus 
  
### Components / Binaries 
- kubectl-gadget
- ig 
- gadgetctl

### (Official) Gadgets

Documentation for the gadgets we provide. Related #2727 and #2374

- Gadget description: What the gadget does, what information it provides, etc. 
- Running the gadget (ig and kubectl-gadget?)
- Gadget parameters
- Kernel requirements

### API 

(#3250)
 
How to use Inspektor Gadget from an API 

- Golang API (examples already present in https://github.com/inspektor-gadget/inspektor-gadget/tree/main/examples/gadgets)
- gRPC API (some things in https://github.com/inspektor-gadget/inspektor-gadget/tree/main/examples/gadgets/grpc/custom_operator)

### Gadget Developers

(#3208)

This section covers the documentation for folks that want to write gadgets.

- Getting started: Quick link that explains how to create a gadget from: https://github.com/inspektor-gadget/inspektor-gadget/blob/main/docs/devel/hello-world-gadget.md
- API
  - For ebpf programs (e.g. gadget_socket_lookup)
  - For wasm programs
- OCI format?
- Gadget metadata
- image build command
  - IG_SOURCE_PATH for in-tree gadgets
- Testing 

### Legacy 

(#3228)

There are still things we need to support for a while, so I'd suggest to move then to a legacy folder

- built-in gadgets
- trace crd

## TBD

- Where to put CLI docs like https://github.com/inspektor-gadget/inspektor-gadget/blob/main/docs/core-concepts/images.md?

### PR DESCRIPTION
**Pull request by mauriciovasquezbernal**
This is a big commit containing a lot of changes to our documentation. The main goal is to make our documentation more focused on the image-based gadgets concept.

Some things done by this commit:

- Update format to docusaurus
- Add Gadget Development
- Add API section (still to be completed)
- Add guides for image-based gadgets
- Remove experimental related stuff from docs

---

This is the result of few PRs that we have merged against the new-docs branch. I'm opening this so folks have the opportunity to run the website locally and check for small mistakes. Feel free to directly push small fixes to this branch (without using --force!!!), I'll squash all commits before merging.

### Testing

Follow the instructions on https://github.com/inspektor-gadget/website/tree/docusaurus to run the website locally. (be sure to checkout the right branches on both, the ig-website and inspektor-gadget repositories)

### TODO

There are still few things that we need to handle, they aren't blockers for the release

- [x] Finish porting gadgets guides
  - [x] snapshot_process #3278 
  - [x] audit _seccomp #3278 
  - [x] top_file #3276
  - [x] top_tcp #3292
  - [x] profile_blockio #3291 
  - [x] snapshot_socket #3295 
- [x] #3277 
- [x] Add operators documentation in reference/operators 
- [ ] Update reference/ig

Fixes #1184

### CODE REVIEW

**Review by mqasimsarfraz — APPROVED** [review]
LGTM, this is exciting! 


One small nit regarding the design guide of prometheus seems outdated should it be moved to legacy?
image

File: docs/api/golang.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+sidebar_position: 20
+description: 'Reference documentation for Inspektor Gadget Golang API'
+---
+
+It's possible to run gadgets and consume their output directly from a Golang
+application. This documentation is still WIP, so please check
+https://github.com/inspektor-gadget/inspektor-gadget/tree/main/examples/gadgets
+to get more information.
+
+TODO: Complete documentation.
```
[the remark this answers is not in the document]
Comment (mauriciovasquezbernal) [reply docs/api/golang.md:12]: Yes, there are still few things missing. I'll open issues for all of them after merging this.

**Comment by mauriciovasquezbernal** [comment]
> Also, I am wondering how do we plant to address the CI failurs for Documentation Checks, particularly see: https://github.com/inspektor-gadget/inspektor-gadget/pull/3273#discussion_r1704272939

It's indeed a chicken-and-egg issue. I think the only option is to merge and fix any possible broken link later on.

File: docs/reference/ig.md [hunk 2]
```diff
@@ -318,7 +318,7 @@ List of flags:
 
 ### Using ig in a Kubernetes pod
 
-In order to run `ig` in a Kubernetes pod use [examples/pod-ig.yaml](examples/pod-ig.yaml).
+In order to run `ig` in a Kubernetes pod use [examples/pod-ig.yaml](../examples/pod-ig.yaml).
```
Comment (mauriciovasquezbernal) [inline docs/reference/ig.md:321]: Ah, I think it's again https://github.com/inspektor-gadget/inspektor-gadget/issues/2254. That change doesn't fix the issue. I think we need to use absolute URLs all around for things in that folder. Let me check it.

File: docs/quick-start.md [hunk 3]
```diff
@@ -4,7 +4,7 @@ sidebar_position: 10
 description: Quick start for running Inspektor Gadget.
 ---
 
-The following examples use the [trace_open](./gadgets/trace_open) Gadget which triggers when a file is open on the system.
+The following examples use the [trace_open](../gadgets/trace_open/README.md) Gadget which triggers when a file is open on the system.
```
Comment (mqasimsarfraz) [inline docs/quick-start.md:7]: Hmm. I don't relative links will work for website. At least it fails for my local rendered version.

Also: https://github.com/inspektor-gadget/inspektor-gadget/pull/2247#discussion_r1407985276

**Comment by mqasimsarfraz** [comment]
> > One small nit regarding the design guide of prometheus seems outdated should it be moved to legacy? or the design section should be hidden?
> 
> It should be hidden https://github.com/inspektor-gadget/website/blob/ad6fe2d3429b36453754954ce2e83dd5fb5904a5/config.yaml#L5, I think that is ignored when building the docs locally.

Okay Perfect!

File: docs/reference/ig.md (hunk 2 above)
Comment (mauriciovasquezbernal) [reply docs/reference/ig.md]: Added a couple of commits that hopefully improve it.

File: docs/quick-start.md (hunk 3 above)
Comment (mauriciovasquezbernal) [reply docs/quick-start.md]: Thanks! Fixed it on my last push!

File: docs/quick-start.md [hunk 4]
```diff
@@ -4,7 +4,7 @@ sidebar_position: 10
 description: Quick start for running Inspektor Gadget.
 ---
 
-The following examples use the [trace_open](../gadgets/trace_open/README.md) Gadget which triggers when a file is open on the system.
+The following examples use the [trace_open](https://github.com/inspektor-gadget/inspektor-gadget/tree/main/gadgets/trace_open) Gadget which triggers when a file is open on the system.
```
[the remark this answers is not in the document]
Comment (mauriciovasquezbernal) [reply docs/quick-start.md]: Pushed a commit updating it.

**Comment by mauriciovasquezbernal** [comment]
Thanks folks for the reviews. Merging it now!

**Comment by burak-ok** [comment]
I was just about to push :D
There is still a broken link. It is fixed in https://github.com/inspektor-gadget/inspektor-gadget/pull/3297

File: gadgets/top_tcp/README.mdx [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+    </TabItem>
+</Tabs>
+
+## Flags
+
+### `--family`
+
+Trace traffic from the specified IPv family. Can be either 4 or 6. Specifying -1 equals to no filtering
+
+Default value: "-1"
```
Comment (alban) [inline gadgets/top_tcp/README.mdx:37]: Did you generate the file manually or did you use README.template? This was moved from README.md to README.mdx but the Makefile still refers to README.md so it seems out of sync.

Comment (mauriciovasquezbernal) [reply gadgets/top_tcp/README.mdx:37]: We manually copied the contents from README.md to README.mdx.

**Pull request merged**